daemon/graphdriver/driver.go

```diff
@@ -114,30 +115,34 @@ func New(root string, options []string) (driver Driver, err error) {
 
 	// Guess for prior driver
 	priorDrivers := scanPriorDrivers(root)
+	if _, ok := priorDrivers["devicemapper"]; ok && len(DefaultDriver) == 0 {
+		return nil, ErrImplicitDeviceMapper
```

Comment:
I'm not sure it's true in all cases: if you are on a RedHat system with a dynamically linked binary, I think it's fine having an automatic election of `devicemapper`.